### DISCUSSION TITLE
Add support for multiple subscriptions to azure service bus scaler

### DISCUSSION THREAD

**Discussion by id27182** [kedacore/keda#1614] [Q&A / Need Help]
**Proposal**

Services, developed by my team is organised in a way, where same service listens for messages from multiple subscriptions, connected to same topic. Each topic have own list of subscriptions with unique names, it can be up to 30 subscriptions for same topic and service. It would be nice if there would be an option to specify multiple subscriptions for same scaler object. It would be perfect if there would be a way to specify willdcard (so all subscriptions will be used for getting metric value). 

**Use-Case**

It will help me to organise my code in more clear and effective manner, without rearchitecting application which I'm working with right now.

**Comment by tomkerkhove** [comment]
Good news - You can already do that today!

Our ScaledObject/ScaledJobs allow you to specify multiple triggers, so in your case, you should add 30 triggers for every individual topic & subscription that needs to be checked.

**Reply by id27182** [reply]
Sorry, I didn't described by case clearly enough: I have a lot of services, each have own topic and certian number of subscriptions. Subscriptions may be created dynamically from code. I'd like to be able to scale based on total number of messages across all subscriptions in topic. From config prospective if can lool like a wildcard ("*") as a value for subscriptionName field. I think if can be easily implemented by updating getSubscriptionEntityFromNamespace() function: I can add there a check for a wildcard in subscriptionName parameter: if subscriptionName is set to "*" - fucntion will list subscriptions  for a topic, get messages count for each of them, and return total number of messages.

**Reply by tomkerkhove** [reply]
Interesting.

That could work, but might give misleading results when misconfiguring it.

Thoughts @ahmelsayed ?

**Reply by ahmelsayed** [reply]
I don't know how service bus is used in that way and if it makes sense to assume wildcard for subscriptions if the name is empty string. Would subscriptionName accepting a comma separated list, and names supporting a `filepath.Match()` style wildcard work better here? 

like 

```yaml
subscriptionName: sub-1, sub-2
# or 
subscriptionName: sub-*, test-*
# or
subscroptionName: '*'

# etc
```

**Reply by id27182** [reply]
definately! 
I've just noticed that wildcard symbol was not displayed in my comment (probably due to markup support or something )

**Reply by tomkerkhove** [reply]
I like the `*` suggestion as the main approach would be to use multiple triggers and I consider this an edge case (don't take it wrong)

**Reply by tomkerkhove** [reply]
Created https://github.com/kedacore/keda/issues/1624, we could also use a list called `subscriptions` that has to be provided if `subscriptionName` is not.

**Reply by tomkerkhove** [reply]
Let's move the conversation there.
